examples/stm32/f4/stm32f4-discovery/lis3dsh_demo/lis3dsh_demo.c

```diff
@@ -0,0 +1,327 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2015 Marco Russi <marco@marcorussi.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+/* ---------------- Inclusions ----------------- */
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/spi.h>
+
+
+
+
+/* ---------------- Local Defines ----------------- */
+
+/* LIS3DSH registers addresses */
+#define ADD_REG_WHO_AM_I				0x0F
+#define ADD_REG_CTRL_4					0x20
+#define ADD_REG_OUT_X_L					0x28
+#define ADD_REG_OUT_X_H					0x29
+#define ADD_REG_OUT_Y_L					0x2A
+#define ADD_REG_OUT_Y_H					0x2B
+#define ADD_REG_OUT_Z_L					0x2C
+#define ADD_REG_OUT_Z_H					0x2D
+
+/* WHO AM I register default value */
+#define UC_WHO_AM_I_DEFAULT_VALUE		0x3F
+
+/* ADD_REG_CTRL_4 register configuration value: X,Y,Z axis enabled and 400Hz of output data rate */
+#define UC_ADD_REG_CTRL_4_CFG_VALUE		0x77
+
+/* Sensitivity for 2G range [mg/digit] */
+#define SENS_2G_RANGE_MG_PER_DIGIT		((float)0.06)
+
+/* LED threshold value in mg */
+#define LED_TH_MG						(1000)	/* 1000mg (1G) */
+
+
+
+
+/* ---------------- Local Macros ----------------- */
+
+/* set read single command. Attention: command must be 0x3F at most */
+#define SET_READ_SINGLE_CMD(x)			(x | 0x80)
+/* set read multiple command. Attention: command must be 0x3F at most */
+#define SET_READ_MULTI_CMD(x)			(x | 0xC0)
+/* set write single command. Attention: command must be 0x3F at most */
+#define SET_WRITE_SINGLE_CMD(x)			(x & (~(0xC0)))
+/* set write multiple command. Attention: command must be 0x3F at most */
+#define SET_WRITE_MULTI_CMD(x)			(x & (~(0x80))	\
```

Comment:
these are only used once, it seems more appropriate (to me at least) to put this sort of thing inside your functions for communicating with the device, rather than separating them out here.  Try and refrain from macros for the sake of it, the compiler is generally very good at inlining things where suitable.